bots/pr/src/main/java/org/openjdk/skara/bots/pr/CheckRun.java

```diff
@@ -104,6 +105,7 @@ private CheckRun(CheckWorkItem workItem, PullRequest pr, Repository localRepo, L
                 workItem.bot.confOverrideName(),
                 workItem.bot.confOverrideRef(),
                 comments);
+        realTargetRef = ApprovalCommand.realTargetRef(pr);
```

Comment:
I would like to avoid calculating this for every pr targeting a pr/X branch. We could make this field lazy initialized in the `needsApproval` method. Something like:

```
if (approval != null) {
    if (realTargetRef == null) {
         realTargetRef = PreIntegration.realTargetRef(pr);
    }
    return approval.needsApproval(realTargetRef);
}
return false;
```

I would also add a comment on the field with something like
```
// Only set if approval is configured for the repo
```

Reply:
Thanks for the suggestion. It will make this patch much better!